lightning/src/ln/channelmanager.rs

```diff
@@ -9038,6 +9038,22 @@ where
 		let reply_paths = self.create_blinded_paths(context)
 			.map_err(|_| Bolt12CreationError::BlindedPathCreationFailed)?;
 
+		let total_liquidity: u64 = self.list_usable_channels().iter().map(|channel| channel.next_outbound_htlc_limit_msat).sum();
```

Comment:
Gotcha, thanks for the tip! I'll make the change.

Reply:
Ahhh shoot! Okay, thanks for your patience :sweat_smile:

Reply:
> `list_channels` is even worse :). Instead iterate over the `Channel`s themselves and call `get_available_balances` on them (see `ChanelDetails::from_channel_context`).

I having trouble figuring out how to iterate directly over the list of channels rather than using the `ChannelDetails` list! Could you point me in the right direction on how to access and iterate over the channels themselves?